write-fonts/src/tables/glyf.rs

```diff
@@ -86,29 +88,291 @@ pub enum BadKurbo {
     HasCubic,
     TooSmall,
     MissingMove,
+    UnequalNumberOfElements(Vec<usize>),
+    InconsistentPathElements(usize, Vec<&'static str>),
+}
+
+/// Point with an associated on-curve flag.
+///
+/// Similar to read_fonts::tables::glyf::CurvePoint, but uses kurbo::Point directly
+/// thus it does not require (x, y) coordinates to be rounded to integers.
+#[derive(Clone, Copy, Debug, PartialEq)]
+struct ContourPoint {
+    point: kurbo::Point,
+    on_curve: bool,
+}
+
+impl ContourPoint {
+    fn new(point: kurbo::Point, on_curve: bool) -> Self {
+        Self { point, on_curve }
+    }
+
+    fn on_curve(point: kurbo::Point) -> Self {
+        Self::new(point, true)
+    }
+
+    fn off_curve(point: kurbo::Point) -> Self {
+        Self::new(point, false)
+    }
+
+    fn distance(&self, other: &Self) -> f64 {
+        self.point.distance(other.point)
+    }
+}
+
+impl From<ContourPoint> for CurvePoint {
+    fn from(pt: ContourPoint) -> Self {
+        let (x, y) = pt.point.ot_round();
+        CurvePoint::new(x, y, pt.on_curve)
+    }
+}
+
+/// A helper struct for building interpolatable contours
+///
+/// Holds a vec of contour, one contour per glyph.
+#[derive(Clone, Debug, PartialEq)]
+struct InterpolatableContourBuilder(Vec<Vec<ContourPoint>>);
+
+impl InterpolatableContourBuilder {
+    /// Create new set of interpolatable contours beginning at the provided points
+    fn new(move_pts: Vec<kurbo::Point>) -> Self {
+        assert!(!move_pts.is_empty());
+        Self(
+            move_pts
+                .into_iter()
+                .map(|pt| vec![ContourPoint::on_curve(pt)])
+                .collect(),
+        )
+    }
+
+    /// Number of interpolatable contours (one per glyph)
+    fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    /// Add a line segment to all contours
+    fn line_to(&mut self, pts: Vec<kurbo::Point>) {
+        assert_eq!(pts.len(), self.len());
+        for (i, pt) in pts.into_iter().enumerate() {
+            self.0[i].push(ContourPoint::on_curve(pt));
+        }
+    }
+
+    /// Add a quadratic curve segment to all contours
+    fn quad_to(&mut self, p0s: Vec<kurbo::Point>, p1s: Vec<kurbo::Point>) {
```

Comment:
Can we take `&[(kurbo::Point, kurbo::Point)]` here instead? It would avoid the need for the zip and the length check.